icu4c/source/common/loclikely.cpp

```diff
@@ -199,24 +199,27 @@ parseTagString(
     goto exit;
 }
 
-#define CHECK_TRAILING_VARIANT_SIZE(trailing, trailingLength) UPRV_BLOCK_MACRO_BEGIN { \
-    int32_t count = 0; \
-    int32_t i; \
-    for (i = 0; i < trailingLength; i++) { \
-        if (trailing[i] == '-' || trailing[i] == '_') { \
-            count = 0; \
-            if (count > 8) { \
-                goto error; \
-            } \
-        } else if (trailing[i] == '@') { \
-            break; \
-        } else if (count > 8) { \
-            goto error; \
-        } else { \
-            count++; \
-        } \
-    } \
-} UPRV_BLOCK_MACRO_END
+namespace {
+inline bool CHECK_TRAILING_VARIANT_SIZE(const char* trailing, int32_t trailingLength) {
+    int32_t count = 0;
+    int32_t i;
+    for (i = 0; i < trailingLength; i++) {
+        if (trailing[i] == '-' || trailing[i] == '_') {
+            count = 0;
+            if (count > 8) {
+                return false;
+            }
+        } else if (trailing[i] == '@') {
```

Comment:
just FYI: I see that we already have fairly low-level code that won't work on EBCDIC machines...